### PR TITLE
Add a scroll restoration test

### PR DESCRIPTION
**Pull request by benmccann**
This test passes on `master`, but fails with https://github.com/sveltejs/kit/pull/3835, which hopefully should help us a bit in preventing regressions in this area.

One thing that's very strange is that it didn't fail on https://github.com/sveltejs/kit/pull/3835 if I navigated to `/anchor/anchor`, but it did if I navigated to `/routing/hashes/target`. I have no explanation for why that would be. It took me a long time to actually figure out how to make the test fail :stuck_out_tongue_closed_eyes:

### CODE REVIEW

File: packages/kit/test/apps/basics/test/test.js [hunk 1]
```diff
@@ -197,7 +197,23 @@ test.describe('Scrolling', () => {
...
+		expect(await page.evaluate(() => scrollY === 0)).toBeTruthy;
+	});
+
+	test('scroll is restored after hitting the back button', async ({ back, clicknav, page }) => {
+		await page.goto('/anchor');
+		await page.click('#scroll-anchor');
+		const originalScrollY = /** @type {number} */ (await page.evaluate(() => scrollY));
+		await clicknav('#routing-page');
+		await back();
+		expect(page.url().substring(page.url().indexOf('/', 8))).toEqual('/anchor#last-anchor-2');
```
Comment (bluwy) [inline packages/kit/test/apps/basics/test/test.js:209]: This is a bit hard to parse, could this be refactored as:
```suggestion
		expect(page.url()).toBe(baseURL + '/anchor#last-anchor-2');
```

The `baseURL` needs to be destructured from the test function.

File: packages/kit/test/apps/basics/test/test.js [hunk 2]
```diff
@@ -197,7 +197,23 @@ test.describe('Scrolling', () => {
...
+		const scrollY = /** @type {number} */ (await page.evaluate(() => scrollY));
+		expect(scrollY).toEqual(originalScrollY);
+		// TODO: fix this. it is failing due to duplicate history entries
+		// https://github.com/sveltejs/kit/issues/3636
+		// await page.goBack();
+		// expect(page.url().substring(page.url().indexOf('/', 8))).toEqual('/anchor');
```
Comment (bluwy) [inline packages/kit/test/apps/basics/test/test.js:215]: And this line too 😬

File: packages/kit/test/apps/basics/test/test.js (hunk 1 above)
Comment (benmccann) [reply packages/kit/test/apps/basics/test/test.js]: thanks. done

File: packages/kit/test/apps/basics/test/test.js (hunk 2 above)
Comment (benmccann) [reply packages/kit/test/apps/basics/test/test.js]: done

File: packages/kit/test/apps/basics/test/test.js [hunk 3]
```diff
@@ -197,7 +197,23 @@ test.describe('Scrolling', () => {
 	}) => {
 		await page.goto('/anchor');
 		await clicknav('#last-anchor-2');
-		expect(await page.evaluate(() => pageYOffset === 0)).toBeTruthy;
+		expect(await page.evaluate(() => scrollY === 0)).toBeTruthy;
```
[the remark this answers is not in the document]
Comment (benmccann) [reply packages/kit/test/apps/basics/test/test.js]: fixed

**Review by bluwy — APPROVED** [review]
